livekit-agents/livekit/agents/tts/tts.py

```diff
@@ -82,7 +89,13 @@ async def aclose(self) -> None:
         self._event_ch.close()
 
     async def __anext__(self) -> SynthesizedAudio:
-        return await self._event_ch.__anext__()
+        if self._timeout > 0:
+            try:
+                return await asyncio.wait_for(self._event_ch.__anext__(), self._timeout)
```

Comment:
asyncio.wait_for is creating a new task for each call to self._event_ch.__anext__().
It seems like we're going to leak one task when the timeout is reached

Reply:
if a timeout occurs then the function inside the `wait_for` gets cancelled immediately before raising `TimeoutError`, so this should be fine